1. **Из колоды в 52 карты извлекаются случайным образом 4 карты.
  a) Найти вероятность того, что все карты – крести.
  б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.**

Используем классическую формулу определения вер-ти:   
$P(A)=\frac{m}{n}$

и формулу сочетаний:
$C^k_n=\frac{n!}{k!(n-k)!}$

Тогда,

a) $m = C^4_{13}=\frac{13!}{4!(13-4)!}$

In [5]:
from math import factorial
def sochetanie(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

print(f'm = {sochetanie(13,4)}')
 

m = 715


$n = C^4_{52}=\frac{52!}{4!(52-4)!}$

In [5]:
print(f'm = {sochetanie(52,4)}')

m = 270725


print(f'P = {715/270725}')

In [11]:
print(f'P = {round(715/270725*100,2)}%')

P = 0.26%


б) "хотя бы один туз" можно найти как обратную вер-ть "из 4-х карт не оказалось ни одного туза" и вычесть эту вер-ть из 1. Тогда,

m = 270725 (найдено выше)

$n = C^4_{48}=\frac{48!}{4!(48-4)!}$ 

In [12]:
print(f'm = {sochetanie(48,4)}')

m = 194580


In [3]:
print(f'P = {round((1-194580/270725)*100,2)}%')

P = 28.13%


2.**На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?**

Количество способов выбрать 3 кнопки из 10 равно числу сочетаний:

$C^3_{10}=\frac{10!}{3!(10-3)!}$

In [6]:
print(f'С = {sochetanie(10,3)}')

С = 120


Тогда вероятность открыть дверь с первой попытки = $\frac{1}{120}$ или 0,83%

3.**В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?**

Используем классическую формулу определения вер-ти:   
$P(A)=\frac{m}{n}$

Тогда,

$m = C^3_{9}=\frac{9!}{3!(9-3)!}$

$n = C^3_{15}=\frac{15!}{3!(15-3)!}$

In [8]:
print(f'm = {sochetanie(9,3)}')
print(f'n = {sochetanie(15,3)}')

m = 84
n = 455


In [12]:
print(f'P = {round(84/455*100,2)}%')

P = 18.46%


4. **В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?**

Вероятность, что первый билет будет выигрышный = $\frac {1}{100}$
Вероятность, что второй билет будет выигрышный = $\frac {1}{99}$
Итоговая вероятность $P = \frac{1}{100} * \frac {1}{99} = \frac {1}{9900}$
или 0,01%